spec/unit/puppet/provider/rabbitmq_parameter/rabbitmqctl_spec.rb

```diff
@@ -52,11 +52,11 @@
 
   it 'matches parameters from list' do
     provider.class.expects(:rabbitmqctl_list).with('parameters', '-p', '/').returns <<-EOT
-shovel  documentumShovel  {"src-uri":"amqp://","src-queue":"my-queue","dest-uri":"amqp://remote-server","dest-queue":"another-queue"}
+shovel  documentumShovel  {"src-uri":["amqp://cl1","amqp://cl2"],"src-queue":"my-queue","dest-uri":"amqp://remote-server","dest-queue":"another-queue"}
 EOT
     expect(provider.exists?).to eq(component_name: 'shovel',
                                    value: {
-                                     'src-uri' => 'amqp://',
+                                     'src-uri' => ['amqp://cl1', 'amqp://cl2'],
```

Comment:
Unless the output is always a list here, rather than change the existing test, I think we need a test for both string and list output formats?

Reply:
Dont we already test for strings in the dest-uri. This isnt hard-coded on the src-uri so this example already covers both

Reply:
Hm. Ok.
I'm still a little confused about where it's a string representing an array and where it's an actual array. Presumably we provide it to Puppet as an array, but when it's passing it to the command line, it's a string formatted like an array, right?

Reply:
That's all in the provider which is using json (parse and to_json). I'm not sure I should test if the json conversion works like it should.

https://github.com/voxpupuli/puppet-rabbitmq/blob/48e8723c55d019ad419a7430700d034a6843fa6a/lib/puppet/provider/rabbitmq_parameter/rabbitmqctl.rb#L20
https://github.com/voxpupuli/puppet-rabbitmq/blob/48e8723c55d019ad419a7430700d034a6843fa6a/lib/puppet/provider/rabbitmq_parameter/rabbitmqctl.rb#L78